php-checks/src/test/resources/checks/ThisVariableUsageInStaticContextCheck.php

```diff
@@ -9,6 +9,21 @@ public static function f() {
 //         ^^^^^
   }
 
+  public static function static_anonymous_function() {
+    $fn = static function() { return $this->field; }; // Noncompliant
+  }
+
+  public static function anonymous_function() {
+    $fn = function() { return $this->field; }; // OK
+    $fn->call(new A());
+  }
+
+  public static function anonymous_function() {
+      $fn = function() { return $this->field; }; // FN - No object gets bound to the closure. To solve this,
```

Comment:
I would not call it a "FN". There's no bug on this line. It all depends on how `$fn` will be used and, as you suggest in your comment, we cannot check that. In this case, as `$fn` is never used, we shouldn't raise an issue.